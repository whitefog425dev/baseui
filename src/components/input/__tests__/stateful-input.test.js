// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StatefulInput, StyledInput} from '../index';

test('StatefulInput - basic render', () => {
  const props = {
    onChange: jest.fn(),
    label: 'Label',
    caption: 'Caption',
    overrides: {
      Input: function CustomInput(props) {
        return (
          <span>
            <StyledInput {...props} />
          </span>
        );
      },
    },
  };

  const component = shallow(<StatefulInput {...props} />);

  expect(component).toMatchSnapshot('renders <StatefulContainer/>');

  expect(component.dive()).toMatchSnapshot('renders <Input/> as a child');
});
