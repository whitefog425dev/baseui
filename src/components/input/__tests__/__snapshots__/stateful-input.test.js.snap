// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatefulInput - basic render: renders <Input/> as a child 1`] = `
<Input
  autoFocus={false}
  caption="Caption"
  disabled={false}
  endEnhancer={null}
  error={false}
  id="bui-mock-id"
  label="Label"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  overrides={
    Object {
      "Input": [Function],
    }
  }
  required={false}
  size="default"
  startEnhancer={null}
  value=""
/>
`;

exports[`StatefulInput - basic render: renders <StatefulContainer/> 1`] = `
<StatefulContainer
  caption="Caption"
  initialState={Object {}}
  label="Label"
  onChange={[MockFunction]}
  overrides={
    Object {
      "Input": [Function],
    }
  }
  stateReducer={[Function]}
/>
`;
